scrape/scrape.go

```diff
@@ -1521,9 +1536,25 @@ loop:
 				targetScrapePoolExceededLabelLimits.Inc()
 				break loop
 			}
+
+			sl.cache.metaMtx.Lock()
+			metaEntry, metaOk := sl.cache.metadata[yoloString(met)]
+			if metaOk {
+				// Originally, the approach in #7771 was to keep lastIterObserved(Type|Unit|Help) fields
+				// and only set the metadata that was specified in _this_ iteration. If any other gields were not
+				// specified, they were treated as zeroed. We have to understand whether we actually need to use this approach
+				// and also take the same approach in the `if ok{` block above.
+				// if metaEntry.lastIterObservedType == currIter {
+				// if metaEntry.lastIterObservedUnit == currIter {
+				// if metaEntry.lastIterObservedHelp == currIter {
+				meta.Type = metaEntry.typ
+				meta.Unit = metaEntry.unit
+				meta.Help = metaEntry.help
+			}
+			sl.cache.metaMtx.Unlock()
 		}
 
-		ref, err = app.Append(ref, lset, t, v)
+		ref, err = app.Append(ref, lset, meta, t, v)
```

Comment:
Hypothetically we only need to append the metadata if the metadata for a series has changed or it's a new series